pkg/R/utils.R

```diff
@@ -218,3 +218,53 @@ sortKeyValueList <- function(kv_list) {
   keys <- sapply(kv_list, function(x) x[[1]])
   kv_list[order(keys)]
 }
+
+## Utility function to generate compact R lists from grouped rdd 
+#  Used in Join-family functions
+genCompactLists <- function(tagged_list, cnull) {
```

Comment:
To increase clarity, could you add some comment on what these two arguments represent and what the function returns? Similarly for `mergeCompactLists` and `joinTaggedList`.